### PR TITLE
epic keyword stuff, show passing pytest assertions in log, `catch_errors` decorator for listeners and suite visitors

### LINKED ISSUE #29: add tests for logging and assert rewriting

### LINKED ISSUE #63: keyword versions of `pytest.raises` and `pytest.warns`
currently they don't show up in the robot log

### LINKED ISSUE #68: `pytest_runtest_protocol` hooks are broken

### PR DESCRIPTION
**Pull request by DetachHead**
- `keywordify` function to allow users to patch 3rd party functions to show as keywords in the log
- patch some pytest methods by default (fixes #63)
- show args in the log for keywords decorated with `@keyword`
- fix `pytest_runtest_protocol` hooks not being called (fixes #68)
- make passed pytest assertions show in the robot log and add tests for pytest assertions (fixes #29)
- add `@catch_errors` decorator

### CODE REVIEW

File: pytest_robotframework/_internal/plugin.py [hunk 1]
```diff
@@ -124,5 +137,6 @@ def pytest_runtest_setup(item: Item):
 def pytest_runtestloop(session: Session) -> object:
     if session.config.option.collectonly:  # type:ignore[no-any-expr]
         return None
+    keywordify(pytest, ["fail", "xfail", "raises", "deprecated_call", "warns"])
```
Comment (DetachHead) [inline pytest_robotframework/_internal/plugin.py:140]: keywordify `BuiltIn` and other robot modules

Comment (DetachHead) [reply pytest_robotframework/_internal/plugin.py:140]: also investigate whether the `xfail` function should be keywordified

Comment (DetachHead) [reply pytest_robotframework/_internal/plugin.py:139]: > keywordify `BuiltIn` and other robot modules

that causes double keywording in robot tests so im leaving it as is for now

> also investigate whether the `xfail` function should be keywordified

yes

File: pytest_robotframework/__init__.py [hunk 2]
```diff
@@ -56,22 +62,246 @@ def import_resource(path: Path | str):
...
+    the `@keyword` decorator is above `@contextmanager`)
+
+    :param name: set a custom name for the keyword in the robot log (default is inferred from the
+    decorated function name). equivalent to `robot.api.deco.keyword`'s `name` argument
+    :param tags: equivalent to `robot.api.deco.keyword`'s `tags` argument
+    :param module: customize the module that appears top the left of the keyword name in the log.
+    defaults to the function's actual module
+    """
+    if fn is None:
+        return _KeywordDecorator(name=name, tags=tags, module=module)
```
Comment (DetachHead) [inline pytest_robotframework/__init__.py:227]: does this still need to be a separate class

**Review by KotlinIsland — APPROVED** [review]

File: pytest_robotframework/__init__.py (hunk 2 above)
Comment (DetachHead) [reply pytest_robotframework/__init__.py:227]: yea it does because the first overload is typed as returning an instance of it

**Pull request merged**